tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,153 @@
+                            "\n# apache::mod::prefork::maxrequestsperchild: 4000" \
+                            "\n# Here are some examples of how you tune the " \
+                            "PostgreSQL options if needed:" \
+                            "\n#" \
+                            "\n# postgresql::server::config_entries:" \
+                            "\n#   max_connections: 600" \
+                            "\n#   shared_buffers: 1024MB" \
+                            "\n# Do not remove " \
+                            "\nmongodb::server::storage_engine: 'wiredTiger'"
+
+
+class ScenarioPerfornaceTuning(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    Performance Tuning utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, we will apply the medium tune size.
+    2. Once the size get apply we will wait for post upgrade test case to check
+    whether the size is same or not.
+    3. In post upgrade we will check whether satellite installer would able to change
+    the size or not
+
+    """
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --tuning medium --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        with open('custom-hiera.yaml', 'w') as fd:
+            fd.write("{}\n".format(DEFAULT_CUSTOM_HIERA_DATA))
+            fd.write("{}".format(MEDIUM_TUNING_DATA))
+        ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+        command_output = [data.strip(" ") for data in
+                          ssh.command('satellite-installer --disable-system-checks')
+                              .stdout]
+        assert "Success!" in command_output
+
+    @post_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In postupgrade scenario we will check the tune paramater state after upgrade
+
+        :id: postupgrade-31e26b08-2157-11ea-9223-001a4a1601d8
+
+        :steps:
+            1. Collect the current tuning state from satellite.yaml file.
+            2. Compare it with the expected value.
+            3. If Expected value get matched then this sceanrio would be passed.
+            1. Run satellite-installer --tuning medium --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        tuning_state_after_upgrade = ssh.command('less /etc/foreman-installer/'
+                                                 'scenarios.d/satellite.yaml|'
+                                                 'grep "tuning: medium";if [ $? -eq 0 ]; '
+                                                 'then echo "PASSED"; else echo "FAILED";'
+                                                 ' fi').stdout
```

Comment:
I find this command strange. Piping less to grep, and printing out string depending on exit code? Just look for string in file and assert on exit code.

```
cmd = 'grep "tuning: medium" /etc/foreman-installer/scenarios.d/satellite.yaml'
tuning_state_after_upgrade = ssh.command(cmd).return_code
assert tuning_state_after_upgrade == 0  # zero indicates success, non-zero indicates error
```

Reply:
Yes, you are correct, my bad I pushed the testing code :)

Reply:
Fixed.